flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobCache.java

```diff
@@ -200,10 +200,14 @@ public void delete(BlobKey key) throws IOException{
 	 * @param key referring to the file to be deleted
 	 */
 	public void deleteGlobal(BlobKey key) throws IOException {
-		delete(key);
 		BlobClient bc = createClient();
-		bc.delete(key);
-		bc.close();
+		delete(key);
```

Comment:
BlobCache.delete can throw IOException (although right now it doesn't) and then `bc` is not closed again. Maybe move it up again and then it should be fine.